gitlab.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/xanzy/go-gitlab"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/storage/memory"
+)
+
+// auditGitlabRepos kicks off audits if --gitlab-user or --gitlab-org options are set.
+// Getting all repositories from the GitLab API and run audit. If an error occurs during an audit of a repo,
+// that error is logged.
+func auditGitlabRepos() ([]Leak, error) {
+	leaks := make([]Leak, 0)
+	repos := make([]*gitlab.Project, 0, 10)
+	page := 1
+	cl := gitlab.NewClient(nil, os.Getenv("GITLAB_TOKEN"))
+
+	// if self hosted GitLab server
+	if url := os.Getenv("GITLAB_URL"); url != "" {
+		cl.SetBaseURL(url)
+	}
+
+	var (
+		ps   []*gitlab.Project
+		resp *gitlab.Response
+		err  error
+	)
+
+	for {
+		if opts.GitLabOrg != "" {
+			opt := &gitlab.ListGroupProjectsOptions{
+				ListOptions: gitlab.ListOptions{
+					PerPage: 10,
```

Comment:
GitLab lets you pull up to 100 records when doing pagination and this would reduce the number of network calls.

```suggestion
					PerPage: 100,
```

Reply:
changed